feature/goal/src/main/kotlin/com/chipichipi/dobedobe/feature/goal/SearchGoalViewModel.kt

```diff
@@ -0,0 +1,53 @@
+package com.chipichipi.dobedobe.feature.goal
+
+import androidx.compose.foundation.text.input.TextFieldState
+import androidx.compose.foundation.text.input.delete
+import androidx.compose.runtime.snapshotFlow
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.chipichipi.dobedobe.core.data.repository.GoalRepository
+import com.chipichipi.dobedobe.core.model.Goal
+import kotlinx.coroutines.flow.SharingStarted
+import kotlinx.coroutines.flow.flatMapLatest
+import kotlinx.coroutines.flow.map
+import kotlinx.coroutines.flow.stateIn
+
+class SearchGoalViewModel(
+    private val goalRepository: GoalRepository,
+) : ViewModel() {
+    val queryState = TextFieldState()
+
+    val uiState =
+        snapshotFlow { queryState.text }
+            .flatMapLatest { query ->
+                goalRepository.getSortedGoals()
+                    .map { goals ->
+                        if (query.isBlank()) {
+                            return@map SearchGoalUiState.Success(goals, emptyList())
+                        }
+
+                        val queriedGoals = goals.filter { query in it.title }
+                        SearchGoalUiState.Success(goals, queriedGoals)
+                    }
+            }
+            .stateIn(viewModelScope, SharingStarted.Lazily, SearchGoalUiState.Loading)
+
+    fun clearQuery() {
+        queryState.edit {
+            delete(0, this.length)
+        }
```

Comment:
```suggestion
    fun clearQuery() {
        queryState.clearText()
    }
```
요건 어떠신가요? 이미 TextFieldState에 있는 동일 동작인 것 같아서요!

Reply:
와우~ clearText() 라는 함수가 있었군요 👍👍 807bddaf9c78c66cae77526f14e5f3c4208f0e41